### ISSUE TITLE
When using Proxying with the ResponseTemplateTransformer the setting proxyUrlPrefixToRemove is ignored

### ISSUE THREAD

**Issue by andrena-eike-kohnert** [wiremock/wiremock#1783]
WireMock version: 2.32.0

# How to reproduce:

Use the following mapping:

```json
{
  "request": {
    "method": "ANY",
    "urlPattern": "/proxy-test/.*"
  },
  "response": {
    "proxyBaseUrl" : "https://localhost:8099",
    "proxyUrlPrefixToRemove" : "/proxy-test"
  }
}
```

Use the setting "--global-response-templating". 

When sending a request to '/proxy-test/foo'
Expected behaviour: a proxy request to https://localhost:8099/foo
Actual behaviour: a proxy request to https://localhost:8099/proxy-test/foo

**Comment by davidseidel** [comment]
I'm facing the same problem, when I try to use this proxy feature. A fix would be great.

**Comment by pyrinoff** [comment]
Same problem. Tryed to create a mock with a mock/intercept pattern. 
First of all, tryed to create proxy: for example, my mock will be at "myserver/smartMock", target request should be "otherserver/original". 

> {

    "mappings": [{
            "request": {
                "urlPath": "/original",
                "method": "POST"
            },
            "response": {
                "body": "ORIGINAL RESPONSE"
            }
        }, {
            "request": {
                "urlPath": "/smartMock",
                "method": "POST"
            },
            "response": {
                "proxyBaseUrl": "http://localhost:8112/original",
                "proxyUrlPrefixToRemove": "/smartMock"
            }
        }
    ]
}


**Result**:          Request was not matched
Closest stub     [path] /smartMock  
Request  /smartMock/original                                 <<<<< URL does not match

/smartMock should route to /original, but /smartMock didnt cut!

Tryed different variants, nothing works for me.
For this variant:
> "proxyBaseUrl": "http://localhost:8112",
request didnt work at all!  Only requests with end "/" is working.

**Comment by grahambunce** [comment]
hi - any idea which release (and associated docker container update) this is fix scheduled for? FYI, documentation refers to this feature so newbies (like me) can get confused when it doesn't work as described :)

**Comment by sergey-karazhenets** [comment]
Still reproduces in version 2.33.2 with option `--global-response-templating`.
But it works with option `--local-response-templating`.

**Comment by balint-backmaker** [comment]
hi @sergey-karazhenets awesome news and thanks for the effort!

Just to be clear, this means when we use `--local-response-templating` it means we need to add `"transformers": ["response-template"]` to the mappings we want to enabled it for, right? I read through the docs a few times over and this is how I interpreted it, but it never says explicitly that `transformers` is directly linked.

The reason for asking is I'm experiencing the same issue and some other flaky behaviour but wanted to test my muppetry thoroughly before I give more feedback on the issue.

Thanks for the answer in advance!

**Comment by balint-backmaker** [comment]
Note to future readers: `--local-response-templating` does not mean that you can use the `proxyUrlPrefixToRemove` in the same mapping as your templated proxy.

here is a pair of working examples, assuming you've ran wiremock with `--local-response-templating`:
```json
{
  "priority": 2,
  "request": {
    "urlPattern": "/.*",
    "headers": {
      "X-WM-PROXY-TARGET": {
        "matches": ".*"
      }
    }
  },
  "response": {
    "proxyBaseUrl": "https://{{request.headers.X-WM-PROXY-TARGET}}.example.com",
    "transformers": ["response-template"],
    "additionalProxyRequestHeaders": {
      "cf-access-token": "{{systemValue type='ENVIRONMENT' key='CF_OPS_TOKEN'}}"
    }
  }
}
```

And then you can do a self-redirect like so:
```json
{
  "priority": 1,
  "request": {
    "urlPattern": "/argo-ops/.*",
    "headers": {
      "X-WM-PROXY-TARGET": {
        "absent": true
      }
    }
  },
  "response": {
    "proxyBaseUrl": "http://localhost:8080",
    "proxyUrlPrefixToRemove": "/argo-ops",
    "additionalProxyRequestHeaders": {
      "X-WM-PROXY-TARGET": "argo-ops"
    }
  }
}
```

I hope this helps! :slightly_smiling_face:

**Comment by tomakehurst** [comment]
Fixed in 9aad8789d2c470eb9e92df83494a2dcffe90ad06

**Comment by tomakehurst** [comment]
Fixed in 9aad8789d2c470eb9e92df83494a2dcffe90ad06
